test/unit/manipulation.js

```diff
@@ -3099,3 +3099,43 @@ testIframe(
 		} );
 	}
 );
+
+QUnit.test( "should handle node removal in event's remove hook (gh-5214)", function( assert ) {
+
+	assert.expect( 4 );
+
+	jQuery(
+		"<div id='container'>" +
+			"<div class='guarded removeself' data-elt='one'>" +
+				"Guarded 1" +
+			"</div>" +
+			"<div class='guarded' data-elt='two'>" +
+				"Guarded 2" +
+			"</div>" +
+			"<div class='guarded' data-elt='three'>" +
+				"Guarded 3" +
+			"</div>" +
+		"</div>"
```

Comment:
I've just noticed this uses different indendation based on the intended HTML indentation. This poses two issues:
1. The final HTML string does not have this indentation at all.
2. If we enable indent ESLint rules for test, this code will be auto-fixed to start in one line.

When doing indentation in HTML strings, indentation should be a part of the string, not done outside. See https://github.com/jquery/jquery/blob/6d78c0768d9aa6ba213678724c89af69a1958df6/test/unit/css.js#L646-L670 for an example of such in-string indentation.

Reply:
Done